test/boost/reader_concurrency_semaphore_test.cc

```diff
@@ -1356,11 +1356,13 @@ memory_limit_table create_memory_limit_table(cql_test_env& env, uint64_t target_
     const auto sstable_write_concurrency = 16;
 
     auto num_sstables = 0;
+    sstables::generation_type gen;
     parallel_for_each(boost::irange(0, sstable_write_concurrency), [&] (int i) {
         return seastar::async([&, i] {
             while (num_sstables != target_num_sstables) {
                 ++num_sstables;
-                auto sst = sst_man.make_sstable(s, sstables_dir.path().string(), sstables::generation_type{num_sstables}, sst_man.get_highest_supported_format(), sstables::sstable_format_types::big);
+                ++gen;
```

Comment:
I think this increment syntax works well only with the current integer based generation, which we want to move away from. So let's make this an explicit method instead: `auto gen = generation_type::generate_random_gen();`
 (name can be something else).

Reply:
We might even rename `generation -> id` to emphasize that this is not a simple incremented integer anymore: it is a unique id, whose actual value has no meaning.
Might not worth the churn though.

Reply:
generation is encoded using an uuid, maybe we can keep this naming legacy? i admit this is not ideal though. but id is way too general. but generation is what we've been using. just like we are still using "disk" even if we are actually talking about is NAND flash memory. strictly speaking, it's wrong. but it's more like a metaphor.

Reply:
Botond is right. Increment syntax would only make sense if we'd enforce a monotonically increasing requirement in the interface. Like it was discussed, sstables::generation_type shouldn't be given the responsibility of generating a new value from the current one.